cube/cube.c

```diff
@@ -4075,13 +4075,21 @@ static void demo_init(struct demo *demo, int argc, char **argv) {
             i++;
             continue;
         }
-        if (strcmp(argv[i], "--width") == 0 && i < argc - 1 && sscanf(argv[i + 1], "%d", &demo->width) == 1 && demo->width > 0) {
-            i++;
-            continue;
+        if (strcmp(argv[i], "--width") == 0 && i < argc - 1) {
+            if (sscanf(argv[i + 1], "%d", &demo->width) == 1 && demo->width > 0) {
+                i++;
+                continue;
+            } else {
+                ERR_EXIT("The --width parameter must be greater than 0", "User Error");
```

Comment:
If the user passes something like `--width max`, the error message will be somewhat misleading.  (I'm not sure this corner case is worth clarifying, however, though I do note that it's not much more code to make the error cases clear...)

Reply:
Also, there's a weird corner case if the user provides `--width` as the last parameter - no error is provoked and the width is silently unchanged.  That may be more of a concern.

Reply:
I've modified the code to respect whether the user has put in a number or not. If they did and its negative, it gives the correct error. If they don't provide a number, it now prints that message.